lib/validation.js

```diff
@@ -105,9 +106,17 @@ function isValidLogger (logger) {
   return result
 }
 
-function schemaCompiler (schema) {
-  return this.ajv.compile(schema)
+function buildSchemaCompiler () {
+  // This instance of Ajv is private
+  // it should not be customized or used
+  const ajv = new Ajv({ coerceTypes: true })
```

Comment:
I think we should support at least these defaults because in most cases you don't need anymore a custom `schemaCompiler`. It also has **no** performance impact.

```
  removeAdditional: true,
  useDefaults:      true,
  coerceTypes:      true,
```